### PR TITLE
[#11492] Support configuring separate front-end URL for E2E testing

### LINKED ISSUE #11492: Support configuring separate front-end URL for E2E testing
Right now, E2E tests assume a single URL for testing, i.e. the back-end and the front-end hosted concurrently.

While it is a reasonable assumption based on the production system's architecture and should remain that way in CI, it is not convenient for development as a simple front-end change necessitates full rebuilt of the front-end, taking (wasting) a few mins time.

Suggestion: make the front-end URL configurable separately for E2E testing. For CI, the value should be the same as the back-end URL.

### PR DESCRIPTION
**Pull request by shadowezz**
Fixes #11492 

**Outline of Solution**

Allow separate configuration of frontend and backend url for E2E testing in `test.properties`. All url navigations uses frontend url but logging out uses backend url.

### CODE REVIEW

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]
Good work! There is just one place that is not correctly changed

File: src/e2e/java/teammates/e2e/util/TestProperties.java [hunk 1]
```diff
@@ -117,7 +121,7 @@ private TestProperties() {
     }
 
     public static boolean isDevServer() {
-        return TEAMMATES_URL.matches("^https?://localhost:[0-9]+(/.*)?");
+        return TEAMMATES_BACKEND_URL.matches("^https?://localhost:[0-9]+(/.*)?");
```
[the remark this answers is not in the document]
Comment (shadowezz) [reply src/e2e/java/teammates/e2e/util/TestProperties.java]: ok i see

File: src/e2e/java/teammates/e2e/cases/SystemErrorEmailReportE2ETest.java [hunk 2]
```diff
@@ -38,7 +38,7 @@ private void testAssertionError() {
 
         ______TS("AssertionError testing");
 
-        String url = createUrl(Const.ResourceURIs.EXCEPTION)
+        String url = createFrontendUrl(Const.ResourceURIs.EXCEPTION)
```
[the remark this answers is not in the document]
Comment (shadowezz) [reply src/e2e/java/teammates/e2e/cases/SystemErrorEmailReportE2ETest.java]: I have made the changes. Just curious what is a resourceUri? Is it the same as an api endpoint?

[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/e2e/java/teammates/e2e/cases/SystemErrorEmailReportE2ETest.java]: That is one way to look at it. RESTful APIs are designed around resources, e.g. `GET /course`, `POST /course`, `DELETE /course` are designed with course as the resource.

[the remark this answers is not in the document]
Comment (shadowezz) [reply src/e2e/java/teammates/e2e/cases/SystemErrorEmailReportE2ETest.java]: ah ok

**Review by wkurniawan07 — APPROVED** [review]
LGTM, thanks for contributing!

**Pull request merged**